.github/workflows/ci.yml

```diff
@@ -59,7 +50,49 @@ jobs:
         shell: bash
 
       - name: Integrations tests
-        run: python -m pytest .github/workflows/tests/
+        run: uv run pytest .github/workflows/tests/
 
       - name: Stop services
         run: docker compose stop
+  
+  deploy:
+    name: Deploy
+    needs: [tests]
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' || github.event_name == 'workflow_dispatch'
```

Comment:
I'm not sure how we want to handle deployments - could deploy on each merge to `main`, new tags, new releases, etc.

Reply:
I think it would make sense to deploy on `tags` so we can have a changelog as well